src/app.rs

```diff
@@ -36,6 +36,7 @@ pub async fn build_with_cert(
     config: config::Config,
     cert_manager: Arc<SecretManager>,
 ) -> anyhow::Result<Bound> {
+    let config = Arc::new(config);
```

Comment:
Can we just `Arc` wrap this in `construct_config`?

Reply:
Yeah I moved it up pretty high. The earlier in the call stack we move it, the less flexibility usages have so I didn't remove the possibility to construct a raw Config without an arc